### PR TITLE
Add gir

### LINKED ISSUE #53: Can't find values to use for gdk_pixbuf::InterpType
gdk_pixbuf::InterpType used to be an enum with the possible values provided but has been replaced by a type definition as i32.  However, I am now unable to determine the valid values for this type in the documentation.  They are needed for methods such as simple_scale().

### PR DESCRIPTION
**Pull request by EPashkin**
Fix #53 

@GuillaumeGomez, @sdroege, @pwil3058

`PixbufFormat` has comment "It's owned by the library and we never free it" but I don't see why.

Don't generated:
`PixbufAnimationIter` etc. can't be fully generated due `glib.TimeVal` (also see #8) and I don't want add `xxManualExt` here.
`Pixbuf` has getters from constant self pointer (same as https://github.com/gtk-rs/glib/pull/249)

### CODE REVIEW

**Comment by EPashkin** [comment]
Generated `Pixbuf`.
Due const IsA errors many getters like `get_colorspace` stays manual, but also have `get_property_colorspace` generated.
Its was not removed to generate `connect_property_colorspace_notify`.

File: src/auto/pixbuf.rs [hunk 1]
```diff
@@ -0,0 +1,531 @@
...
+    }
+
+    #[cfg(any(feature = "v2_32", feature = "dox"))]
+    pub fn new_from_bytes(data: &glib::Bytes, colorspace: Colorspace, has_alpha: bool, bits_per_sample: i32, width: i32, height: i32, rowstride: i32) -> Pixbuf {
+        unsafe {
+            from_glib_full(ffi::gdk_pixbuf_new_from_bytes(data.to_glib_none().0, colorspace.to_glib(), has_alpha.to_glib(), bits_per_sample, width, height, rowstride))
+        }
+    }
+
+    //pub fn new_from_data<'a, P: Into<Option<&'a /*Unimplemented*/PixbufDestroyNotify>>, Q: Into<Option</*Unimplemented*/Fundamental: Pointer>>>(data: &[u8], colorspace: Colorspace, has_alpha: bool, bits_per_sample: i32, width: i32, height: i32, rowstride: i32, destroy_fn: P, destroy_fn_data: Q) -> Pixbuf {
```
Comment (sdroege) [inline src/auto/pixbuf.rs]: This one could be (manually) implemented for anything that is `AsRef<u8>` or `AsMut<u8>` btw (depending on whether the pixbuf is allowed to modify the underlying data or not). See e.g. https://github.com/sdroege/gstreamer-rs/blob/963557b79f5d3a91db944ed953a8c9fcab78a6ba/gstreamer/src/buffer.rs#L100

File: src/auto/pixbuf.rs [hunk 2]
```diff
@@ -0,0 +1,531 @@
...
+        }
+    }
+
+    pub fn get_formats() -> Vec<PixbufFormat> {
+        unsafe {
+            FromGlibPtrContainer::from_glib_container(ffi::gdk_pixbuf_get_formats())
+        }
+    }
+
+    //pub fn new_from_stream_async<'a, P: IsA</*Ignored*/gio::InputStream>, Q: Into<Option<&'a gio::Cancellable>>, R: Fn(Result<(), Error>) + Send + Sync + 'static>(stream: &P, cancellable: Q, callback: R) {
```
Comment (sdroege) [inline src/auto/pixbuf.rs]: gio::InputStream, OutputStream and IOStream should be generatable at this point

File: src/auto/pixbuf.rs [hunk 3]
```diff
@@ -0,0 +1,531 @@
...
+
+    #[cfg(any(feature = "v2_36", feature = "dox"))]
+    fn copy_options(&self, dest_pixbuf: &Pixbuf) -> bool;
+
+    fn fill(&self, pixel: u32);
+
+    fn get_option(&self, key: &str) -> Option<String>;
+
+    //#[cfg(any(feature = "v2_32", feature = "dox"))]
+    //fn get_options(&self) -> /*Unknown conversion*//*Unimplemented*/HashTable TypeId { ns_id: 0, id: 28 }/TypeId { ns_id: 0, id: 28 };
```
Comment (sdroege) [inline src/auto/pixbuf.rs]: Not sure why this is not generated. It's a `HashMap<String, String>` that would be needed here, which we cover in glib's translate.rs

File: src/auto/pixbuf.rs (hunk 1 above)
Comment (EPashkin) [reply src/auto/pixbuf.rs]: I don't like add manual new function at this stage

File: src/auto/pixbuf.rs [hunk 4]
```diff
@@ -0,0 +1,531 @@
...
+
+    fn new_subpixbuf(&self, src_x: i32, src_y: i32, width: i32, height: i32) -> Option<Pixbuf>;
+
+    #[cfg(any(feature = "v2_36", feature = "dox"))]
+    fn remove_option(&self, key: &str) -> bool;
+
+    //fn save<'a, P: Into<Option<&'a Error>>>(&self, filename: &str, type_: &str, error: P, : /*Unknown conversion*//*Unimplemented*/Fundamental: VarArgs) -> bool;
+
+    //fn save_to_buffer<'a, P: Into<Option<&'a Error>>>(&self, type_: &str, error: P, : /*Unknown conversion*//*Unimplemented*/Fundamental: VarArgs) -> Option<Vec<u8>>;
```
Comment (sdroege) [inline src/auto/pixbuf.rs]: These two and the loading variants should be easy to cover too

**Comment by sdroege** [comment]
Looks generally good to me

File: src/auto/pixbuf.rs (hunk 3 above)
Comment (EPashkin) [reply src/auto/pixbuf.rs]: Added issue to remember this https://github.com/gtk-rs/gir/issues/507

File: src/auto/pixbuf.rs (hunk 2 above)
Comment (EPashkin) [reply src/auto/pixbuf.rs]: Your right, at minimum `gio::InputStream`, seems fully generated.
Same, IMHO this not part of this PR.

File: src/auto/pixbuf.rs (hunk 4 above)
Comment (EPashkin) [reply src/auto/pixbuf.rs]: IMHO https://developer.gnome.org/gdk-pixbuf/stable/gdk-pixbuf-File-saving.html#gdk-pixbuf-save unusable without proper vararg support

File: src/auto/pixbuf.rs (hunk 2 above)
Comment (EPashkin) [reply src/auto/pixbuf.rs]: I preparing PR for Gio

**Comment by GuillaumeGomez** [comment]
For me as well so let's merge.

@EPashkin: Please add the missing implementation that @sdroege pointed.

File: src/auto/pixbuf.rs (hunk 4 above)
Comment (sdroege) [reply src/auto/pixbuf.rs]: You can manually implement it, and just always pass NULL for the varargs

**Comment by sdroege** [comment]
All good for me to go btw, I was just pointing out things that could be improved on top of this

File: src/auto/pixbuf.rs (hunk 4 above)
Comment (GuillaumeGomez) [reply src/auto/pixbuf.rs]: @EPashkin: What's missing with the [va_list-rs](https://crates.io/crates/va_list-rs) crate?

Comment (EPashkin) [reply src/auto/pixbuf.rs]: I still even don't looked at it :(

Comment (GuillaumeGomez) [reply src/auto/pixbuf.rs]: Oh. :'(

**Pull request merged**